openpype/modules/kitsu/actions/launcher_show_in_kitsu.py

```diff
@@ -0,0 +1,130 @@
+import webbrowser
+
+from openpype.pipeline import LauncherAction
+from openpype.modules import ModulesManager
+from openpype.client import get_project, get_asset_by_name
+
+
+class ShowInKitsu(LauncherAction):
+    name = "showinkitsu"
+    label = "Show in Kitsu"
+    icon = "external-link-square"
+    color = "#e0e1e1"
+    order = 10
+
+    @staticmethod
+    def get_kitsu_module():
+        return ModulesManager().modules_by_name.get("kitsu")
+
+    def is_compatible(self, session):
+        if not session.get("AVALON_PROJECT"):
+            return False
+
+        return True
+
+    def process(self, session, **kwargs):
+
+        # Context inputs
+        project_name = session["AVALON_PROJECT"]
+        asset_name = session.get("AVALON_ASSET", None)
+        task_name = session.get("AVALON_TASK", None)
+
+        project = get_project(project_name=project_name,
+                              fields=["data.zou_id"])
+        if not project:
+            raise RuntimeError(f"Project {project_name} not found.")
+
+        project_zou_id = project["data"].get("zou_id")
+        if not project_zou_id:
+            raise RuntimeError(f"Project {project_name} has no "
+                               f"connected ftrack id.")
+
+        asset_zou_data = None
+        task_zou_id = None
+        asset_zou_name = None
+        asset_zou_id = None
+        asset_zou_type = 'Assets'
+        zou_sub_type = ['AssetType', 'Sequence']
+        if asset_name:
+            asset_zou_name = asset_name
+            asset_fields = ["data.zou.id", "data.zou.type"]
+            if task_name:
+                asset_fields.append(f"data.tasks.{task_name}.zou.id")
+
+            asset = get_asset_by_name(project_name,
+                                      asset_name=asset_name,
+                                      fields=asset_fields)
+
+            asset_zou_data = asset["data"].get("zou")
+
+            if asset_zou_data:
+                asset_zou_type = asset_zou_data["type"]
+                if asset_zou_type not in zou_sub_type:
+                    asset_zou_id = asset_zou_data["id"]
+            else:
+                asset_zou_type = asset_name
```

Comment:
This code is hard to grasp - why would the **type** be set to the asset name? Is this a bug?

Reply:
If the selected asset is not a task with an ID, then the returned value is the "string" of the element selected from OP.
This value is then used to know what to filter in Kitsu.

Reply:
Ah - I see! That's because Kitsu syncs with the `Assets/` and `Shots/` folders explicitly into OpenPype. And you're doing a straight conversion of that name to type.

Could you please add a comment into the code to explain this logic/behavior?